pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -135,21 +145,47 @@ func configureFromProviderConfig(pushCtx *model.PushContext, meta *model.NodeMet
 
 	switch provider := providerCfg.Provider.(type) {
 	case *meshconfig.MeshConfig_ExtensionProvider_Zipkin:
-		tracing, err = buildHCMTracing(pushCtx, providerCfg.Name, provider.Zipkin.Service, provider.Zipkin.Port, provider.Zipkin.MaxTagLength, zipkinConfigGen)
+		tracing, err = buildHCMTracing(pushCtx, envoyZipkin, provider.Zipkin.Service, provider.Zipkin.Port, provider.Zipkin.MaxTagLength, zipkinConfigGen)
 	case *meshconfig.MeshConfig_ExtensionProvider_Datadog:
-		tracing, err = buildHCMTracing(pushCtx, providerCfg.Name, provider.Datadog.Service, provider.Datadog.Port, provider.Datadog.MaxTagLength, datadogConfigGen)
+		tracing, err = buildHCMTracing(pushCtx, envoyDatadog, provider.Datadog.Service, provider.Datadog.Port, provider.Datadog.MaxTagLength, datadogConfigGen)
 	case *meshconfig.MeshConfig_ExtensionProvider_Lightstep:
-		tracing, err = buildHCMTracing(pushCtx, providerCfg.Name, provider.Lightstep.Service, provider.Lightstep.Port, provider.Lightstep.MaxTagLength,
-			func(hostname, clusterName string) (*anypb.Any, error) {
-				lc := &tracingcfg.LightstepConfig{
-					CollectorCluster: clusterName,
-					AccessTokenFile:  provider.Lightstep.AccessToken,
-				}
-				return anypb.New(lc)
-			})
+		// Envoy dropped support for Lightstep in v1.24+ (~istio 1.16+). So, we will generate old-style configuration
+		// for lightstep for the last 5 versions of Istio (covering support window), but OTel-based configuration for all other cases
+		if strings.HasPrefix(meta.IstioVersion, "1.11") || strings.HasPrefix(meta.IstioVersion, "1.12") ||
```

Comment:
prefer `util.IsIstioVersionGE114` and similar functions (or make your own if needed)

Reply:
Then its just `IsGE16 { otel } else {lightstep}` I think?

Reply:
i was not aware of those methods. That should work. @kyessenov's proposal is probably a better solution if we want to pursue extension-awareness.